File: algorithm/487. Max Consecutive Ones II/487.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int max=0;
        int tem=0;
        vector<int> a;
        for (int i=0;i<nums.size();i++)
        {
            if(nums[i]==1)
            {
                tem++;
            }
            else
            {
                if(tem!=0)
                    a.push_back(tem);
                a.push_back(0);
                tem=0;
            }
        }
        if(tem!=0)
            a.push_back(tem);
        for (int i =0;i<a.size();i++)
        {
            if(i+2<a.size())
                a[i]=a[i]+a[i+1]+a[i+2]+1;
            else if(i+1<a.size())
                a[i]=a[i]+a[i+1]+1;
            else if(a[i]==0)
                a[i]=1;
        }
        sort(a.begin(),a.end());
        return a[a.size()-1];
        
    }
};

File: algorithm/487. Max Consecutive Ones II/487.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int res = 0, zeroLeft = 0, zeroRight = 0;
        for(int i = 0; i < nums.length; i++) {
            zeroRight++;
            if(nums[i] == 0) {
                zeroLeft = zeroRight;
                zeroRight = 0;
            }
            res = Math.max(res, zeroLeft + zeroRight);
        }
        return res;
    }
}
